go/internal/feast/registry/registry.go

```diff
@@ -35,24 +37,25 @@ type Registry struct {
 	cachedRegistry                 *core.Registry
 	cachedRegistryProtoLastUpdated time.Time
 	cachedRegistryProtoTtl         time.Duration
-	mu                             sync.Mutex
+	mu                             sync.RWMutex
 }
 
-func NewRegistry(registryConfig *RegistryConfig, repoPath string) (*Registry, error) {
+func NewRegistry(registryConfig *RegistryConfig, repoPath string, project string) (*Registry, error) {
 	registryStoreType := registryConfig.RegistryStoreType
 	registryPath := registryConfig.Path
 	r := &Registry{
-		cachedRegistryProtoTtl: time.Duration(registryConfig.CacheTtlSeconds),
+		project:                project,
+		cachedRegistryProtoTtl: time.Duration(registryConfig.CacheTtlSeconds) * time.Second,
```

Comment:
This seems like a pretty large change in time proportional to the number of seconds, no? Was this on purpose?

Reply:
This one may need @EXPEbdodla 's help to explain.